### PR TITLE
Bug 1796040: pkg/manifests: change location of trustedCA bundle; add trustedCA to …

### PR DESCRIPTION
**Pull request by lilic**
…alertmanager oauth-proxy

### CODE REVIEW

**Comment by s-urbaniak** [comment]
so far lgtm :+1:

**Comment by lilic** [comment]
/retest

flake

**Comment by lilic** [comment]
@s-urbaniak PTAL, thanks!

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -331,6 +331,11 @@ func (f *Factory) AlertmanagerMain(host string, trustedCABundleCM *v1.ConfigMap)
 
 	if trustedCABundleCM != nil {
 		volumeName := "alertmanager-trusted-ca-bundle"
+		for in, container := range a.Spec.Containers {
```
Comment (s-urbaniak) [inline pkg/manifests/manifests.go]: nit: isn't the container index deterministic. we rely on this invariant in a couple of other places.

Comment (lilic) [reply pkg/manifests/manifests.go]: I would not like to rely on an order of the containers, even if we are doing it other places. Don't see a problem with this current implementation. I would even suggestion to not make it deterministic, as its bad practice IMHO, as we clearly all don't know that. :D

**Pull request merged**